python/pyspark/sql/types.py

```diff
@@ -92,6 +92,23 @@ def fromInternal(self, obj):
         """
         return obj
 
+    @Since(3.0)
+    def toDDL(self):
+        """
+        Returns a string containing a schema in DDL format.
+        """
+        sc = SparkContext._active_spark_context
+        dt = sc._jvm.__getattr__("org.apache.spark.sql.types.DataType$").__getattr__("MODULE$")
+        json = self.json()
+        return dt.fromJson(json).toDDL()
```

Comment:
An alternative approach would be to call self._jdf.schema().toDDL(), similar to dataframe.printSchema(). This would diverge the APIs of Scala/Python but keep the implementation in Scala.